webapp/loggedin/appHeader/components/cycleSelector.js

```diff
@@ -0,0 +1,34 @@
+import React from 'react'
+import * as R from 'ramda'
+
+import Survey from '../../../../common/survey/survey'
+
+import { useI18n } from '../../../commonComponents/hooks'
+import Dropdown from '../../../commonComponents/form/dropdown'
+
+const CycleSelector = props => {
+  const { surveyInfo, surveyCycle, onChange } = props
+  const cycleKeys = R.pipe(Survey.getCycles, R.keys)(surveyInfo)
+
+  if (cycleKeys.length === 1)
+    return null
+
+  const i18n = useI18n()
+
+  const cycleItems = cycleKeys.map(key =>
+    ({ key, value: `${i18n.t(`common.cycle`)} ${Number(key) + 1}` })
+  )
+  const cycleSelection = R.find(R.propEq('key', surveyCycle), cycleItems)
+
+  return (
+    <Dropdown
+      items={cycleItems}
+      selection={cycleSelection}
```

Comment:
you can use Dropdonw itemKeyProp="key" and pass surveyCycle directly to selection

Reply:
not really. surveyCycle is just the key, not the item itself. I have renamed the variable to surveyCycleKey for better clarity